Add a cmake file to properly handle the source files.

tastytea opened the pull request:
This is [the Debian patch](https://sources.debian.org/src/cld2/0.0.0-git20150806-9/debian/patches/add-cmake-file.patch/) which is apparently based on the patch from <https://github.com/CLD2Owners/cld2/issues/29> that isn't accessible anymore. I've added `set(CMAKE_CXX_STANDARD 98)` and changed the installation directory of the libraries to `${CMAKE_INSTALL_LIBDIR}`.

----

Changes from to the original patch:
  - Set the standard to C++98
  - Use correct installation directory for libraries

The last recorded SVN revision in the git log is 196, Debian uses
0.0.197 as version for their 2015-08-06 snapshot. The only change since
then was a typo in the readme, so I'm keeping the version.

Author: Gianfranco Costamagna
Forwarded: https://github.com/CLD2Owners/cld2/issues/29
Forwarded: https://sources.debian.org/src/cld2/0.0.0-git20150806-9/debian/patches/add-cmake-file.patch/
Closes: https://github.com/CLD2Owners/cld2/issues/29

tastytea committed:
Add a cmake file to properly handle the source files.

Changes from to the original patch:
  - Set the standard to C++98
  - Use correct installation directory for libraries

The last recorded SVN revision in the git log is 196, Debian uses
0.0.197 as version for their 2015-08-06 snapshot. The only change since
then was a typo in the readme, so I'm keeping the version.

Author: Gianfranco Costamagna <costamagnagianfranco@yahoo.it>
Forwarded: https://github.com/CLD2Owners/cld2/issues/29
Forwarded: https://sources.debian.org/src/cld2/0.0.0-git20150806-9/debian/patches/add-cmake-file.patch/
Closes: https://github.com/CLD2Owners/cld2/issues/29

ryonakano commented on CMakeLists.txt:
```diff
@@ -0,0 +1,158 @@
+cmake_minimum_required(VERSION 3.0)
```
```suggestion
cmake_minimum_required(VERSION 3.10)
```

So that we can build with CMake >= 4.0

https://cmake.org/cmake/help/latest/release/4.0.html#deprecated-and-removed-features

tastytea committed:
bump minimum CMake version to 3.10

So that we can build with CMake >= 4.0
https://cmake.org/cmake/help/latest/release/4.0.html#deprecated-and-removed-features

Co-authored-by: Ryo Nakano <ryonakaknock3@gmail.com>
